### PR TITLE
Fix an s_server infinite loop

### PR DESCRIPTION
**Pull request by mattcaswell**
Commit c4666bfa changed s_server so that it asked libssl rather than the
underlying socket whether an error is retryable or not on the basis that
libssl has more information. That is true unfortunately the method used
was wrong - it only checks libssl's own internal state rather than both
libssl and the BIO. Should use SSL_get_error() instead.

This issue can cause an infinite loop because some errors could appear as
retryable when in fact they are not.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: apps/s_server.c [hunk 1]
```diff
@@ -2606,6 +2606,15 @@ static void close_accept_socket(void)
     }
 }
 
+static int is_retryable(SSL *con, int i)
+{
+    int err = SSL_get_error(con, i);
+
+    /* If it's not a fatal error, it must be retryable */
+    return (err != SSL_ERROR_SSL) && (err != SSL_ERROR_SYSCALL)
```
Comment (richsalz) [inline apps/s_server.c]: I prefer each "arm" of the ```&&``` to be separate

Comment (mattcaswell) [inline apps/s_server.c]: By separate, you mean on a separate line?

Comment (richsalz) [inline apps/s_server.c]: yes.  like your cascading packet/wpacket calls, or the TEST calls with the new framework.

Comment (mattcaswell) [inline apps/s_server.c]: Ok. Fixed. New commit pushed.

Comment (levitte) [inline apps/s_server.c]: Oy, how `util/openssl-format-source` is gonna change things when / if we run it again...

Not complaining about the result here, but let's be aware how we change the coding style be sheer habits.

**Comment by mattcaswell** [comment]
Ping @richsalz

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**